docs/content/concepts/sui-move-concepts/conventions.mdx

```diff
@@ -0,0 +1,331 @@
+  // ✅ Right
+  const ENameHasMaxLengthOf64Chars: u64 = 0;
+
+  // ❌ Wrong
+  const INVALID_NAME: u64 = 0;
+}
+```
+
+## Struct Property Comments
+
+#### Describe the properties of your structs.
+
+```move
+module conventions::profile {
+  struct Profile has key, store {
+    id: UID,
+    // The age of the user
+    age: u8,
+    // The first name of the user
+    name: String
+  }
+}
+```
+
+## Destroy Functions
+
+#### Provide functions to delete objects and structs. Empty objects must be destroyed with the function destroy_empty. The function drop must be used for objects that have types that can be dropped.
+
+```move
+module conventions::wallet {
+  struct Wallet<Value> {
+    id: UID,
+    value: Value
+  }
+
+  // Value has drop
+  public fun drop<Value: drop>(wallet: &mut Wallet<Value>) {}
+
+  // Value doesn't have drop
+  // Does NOT throw if the `wallet` is not empty.
+  public fun destroy<Value>(wallet: &mut Wallet<Value>) {}
+
+  // Value doesn't have drop
+  // Throws if the `wallet` is not empty.
+  public fun destroy_empty<Value>(wallet: &mut Wallet<Value>) {}
+}
+```
+
+## Pure Functions
+
+#### Keep your functions pure to maintain composability. Do not use transfer::transfer or transfer::public_transfer inside core functions.
+
+```move
+module conventions::amm {
+  struct Pool has key {
+    id: UID
+  }
+
+  // ✅ Right
+  // Return the excess coins even if they have zero value.
+  public fun add_liquidity<CoinX, CoinY, LP_Coin>(pool: &mut Pool, coin_x: Coin<CoinX>, coin_y: Coin<CoinY>): (Coin<LpCoin>, Coin<CoinX>, Coin<CoinY>) {}
+
+  // ✅ Right
+  public fun add_liquidity_and_transfer<CoinX, CoinY, LP_Coin>(pool: &mut Pool, coin_x: Coin<CoinX>, coin_y: Coin<CoinY>, recipient: address) {
+    let (lp_coin, coin_x, coin_y) = add_liquidity(pool, coin_x, coin_y);
+    transfer::public_transfer(lp_coin, recipient);
+    transfer::public_transfer(coin_x, recipient);
+    transfer::public_transfer(coin_y, recipient);
+  }
+
+  // ❌ Wrong
+  public fun add_liquidity<CoinX, CoinY, LP_Coin>(pool: &mut Pool, coin_x: Coin<CoinX>, coin_y: Coin<CoinY>, ctx: &mut TxContext): Coin<LpCoin> {
+    transfer::public_transfer(coin_x, tx_context::sender(ctx));
+    transfer::public_transfer(coin_y, tx_context::sender(ctx));
+
+    coin_lp
+  }
+}
+```
+
+## Coin argument
+
+#### Pass the Coin by value with the right amount directly, it's better for transaction readability from the frontend.
+
+```move
+module conventions::amm {
+  struct Pool has key {
+    id: UID
+  }
+
+  // ✅ Right
+  public fun swap<CoinX, CoinY>(coin_in: Coin<CoinX>): Coin<CoinY> {}
+
+  // ❌ Wrong
+  public fun swap<CoinX, CoinY>(coin_in: &mut Coin<CoinX>): Coin<CoinY> {}
+}
+```
+
+## Account Information
+
+#### To maintain composability, do not store the user's account data in a hashmap - e.g. (Table/Bag/VecSet). Create an object and return it to the user. Moreover, parallelization on Sui depends on objects so it's always a good idea to split the app state to a maximum.
```

Comment:
I don't think it should be that restrictive. Many applications would want to keep access to user data so they can mutate it. However, this line of thinking leads to two different points:

1. Don't perform authorization based on address, use capability (perhaps, worth making a separate section?)
2. If there's user data that doesn't need to be managed in a centralized fashion, it's better to store it in an owned object.

Reply:
We should break down into two sections!